### PR TITLE
user_db-layer WIP

### PR DESCRIPTION
**Pull request by MarianDropa**

### CODE REVIEW

File: src/main/java/dao/ProductDao.java [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
- */
+import model.Product;
+
+import java.util.ArrayList;
+
 public interface ProductDao {
+
+    void addProduct(String productName, String firm, Integer price, Integer maxLoad, String loadType);
+
+    void save(ArrayList<Product> productdb);
```
Comment (lazermann111) [inline src/main/java/dao/ProductDao.java:11]: ArrayList -> List

File: src/main/java/dao/ProductDao.java [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+
+    Product getByName(String productName);
+
+    Product getById(Integer productId);
+
+    ArrayList<Product> getAll();
```
Comment (lazermann111) [inline src/main/java/dao/ProductDao.java:17]: ArrayList -> List

File: src/main/java/dao/ProductDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
+package dao;
+
+import model.Product;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+public class ProductDaoImpl implements ProductDao{
+    static ArrayList<Product> productdb = new ArrayList<>();
```
Comment (lazermann111) [inline src/main/java/dao/ProductDaoImpl.java:10]: ArrayList -> List

File: src/main/java/dao/ProductDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        } catch (IOException e) {
+            System.out.println("Error occurred while saving user");
+        }
+    }
+    }
+
+    @Override
+    public Product getByName(String productName) {
+        for (Product product : productdb) {
+            if (product.getProductName().equals(productName)) {
```
Comment (lazermann111) [inline src/main/java/dao/ProductDaoImpl.java:62]: it seems that we can have multiple products with same name, maybe we need to return List here then?

File: src/main/java/dao/ProductDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        for (Product product : productdb) {
+            if (product.getProductId().equals(productId)) {
+                return product;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public ArrayList<Product> getAll() {
```
Comment (lazermann111) [inline src/main/java/dao/ProductDaoImpl.java:80]: ArrayList -> List

File: src/main/java/dao/UserDao.java [hunk 6]
```diff
@@ -1,7 +1,25 @@
...
+import java.util.List;
+
 /**
  * Created by Igor on 10/8/2019.
  */
 public interface UserDao {
+
+    void addUser(String username, String password);
+
+    void save(ArrayList<User> userdb);
```
Comment (lazermann111) [inline src/main/java/dao/UserDao.java:15]: ArrayList -> List

File: src/main/java/dao/UserDao.java [hunk 7]
```diff
@@ -1,7 +1,25 @@
...
+
+    void isLocked(Integer userId, boolean isLocked);
+
+    User getByName(String username);
+
+    User getById(Integer userId);
+
+    ArrayList<User> getAll();
```
Comment (lazermann111) [inline src/main/java/dao/UserDao.java:23]: ArrayList -> List

File: src/main/java/dao/UserDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+import model.User;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+
+public class UserDaoImpl implements UserDao {
+
+    static ArrayList<User> userdb = new ArrayList<>();
```
Comment (lazermann111) [inline src/main/java/dao/UserDaoImpl.java:12]: ArrayList -> List

File: src/main/java/dao/UserDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+
+    @Override
+    public void save(ArrayList<User> userdb) {
```
Comment (lazermann111) [inline src/main/java/dao/UserDaoImpl.java:16]: ArrayList -> List

File: src/main/java/dao/UserDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+    @Override
+    public void addUser(String username, String password) {
+
+        userdb.add(new User(username, password));
+        save(userdb);
+    }
+
+    @Override
+    public void isLocked(Integer userId, boolean isLocked) {
```
Comment (lazermann111) [inline src/main/java/dao/UserDaoImpl.java:58]: this method is setter (cause it sets value), so better to rename it (lockUser/setLocked etc)

**Review by lazermann111 — COMMENTED** [review]
plz see comments

File: src/main/java/dao/ProductDaoImpl.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
+package dao;
+
+import model.Product;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+public class ProductDaoImpl implements ProductDao{
```
Comment (lazermann111) [inline src/main/java/dao/ProductDaoImpl.java:9]: There is no methdo for loading products from file into collection on app startup, plz add it

File: src/main/java/dao/UserDaoImpl.java [hunk 12]
```diff
@@ -0,0 +1,93 @@
+package dao;
+
+import model.User;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+
+public class UserDaoImpl implements UserDao {
```
Comment (lazermann111) [inline src/main/java/dao/UserDaoImpl.java:10]: There is no method for loading users from file into collection on app startup, plz add it

File: src/main/java/dao/ProductDao.java (hunk 1 above)
Comment (olyarozhko) [reply src/main/java/dao/ProductDao.java:11]: addProduct() should have "Product newProduct" as a parameter

File: src/main/java/dao/UserDao.java (hunk 6 above)
Comment (olyarozhko) [reply src/main/java/dao/UserDao.java:15]: void addUser() should have "User userNew" as a parameter

File: src/main/java/dao/OrderDao.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
+package dao;
+
+import model.Order;
+
+import java.util.List;
+
+public interface OrderDao {
+
+    void readFromCsv(String filename);
```
Comment (lazermann111) [inline src/main/java/dao/OrderDao.java:9]: this is method specific to FileOrderDaoImpl, its N/A to other dao (in-mem/db etc), plz remove it from here

File: src/main/java/dao/OrderDaoImpl.java [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+            if (order.getId().equals(id)) {
+                return order;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public List<Order> getAll() {
+        return (orderdb);
```
Comment (lazermann111) [inline src/main/java/dao/OrderDaoImpl.java:106]: why (orderdb) ? :)

File: src/main/java/dao/ProductDao.java [hunk 15]
```diff
@@ -1,7 +1,22 @@
...
-/**
- * Created by Igor on 10/8/2019.
- */
+import model.Product;
+
+import java.util.List;
+
 public interface ProductDao {
+
+    void readFromCsv(String filename);
```
Comment (lazermann111) [inline src/main/java/dao/ProductDao.java:9]: same, remove this method from interface

File: src/main/java/dao/UserDao.java [hunk 16]
```diff
@@ -1,7 +1,26 @@
...
+import model.User;
+
+import java.util.List;
+
 /**
  * Created by Igor on 10/8/2019.
  */
 public interface UserDao {
+
+    void readFromCsv(String filename);
```
Comment (lazermann111) [inline src/main/java/dao/UserDao.java:12]: same

**Pull request merged**